lib/AgrirouterClient/src/ConnectionProvider/MqttConnectionProvider.cpp

```diff
@@ -0,0 +1,114 @@
+#include "MqttConnectionProvider.h"
+
+#include "AgrirouterClient.h"
+#include "AgrirouterMessageUtils.h"
+#include "Utils.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+
+MqttConnectionProvider::MqttConnectionProvider(Settings *settings)
+{
+    m_settings = settings;
+    m_body = "";
+    m_url = "";
+    m_mqttClient = NULL;
+    this->init();
+}

```

Comment:
Better initialize member variables in header file or in initializer list.
This prevents double initialization because the compiler does default inits before entering the constructor (if there is no init in header file or in initializer list).